chips/nrf51/src/ble_advertising_driver.rs

```diff
@@ -196,29 +213,51 @@ impl<'a, A: hil::time::Alarm + 'a> BLE<'a, A> {
     }
 }
 
-impl<'a, A: hil::time::Alarm + 'a> hil::time::Client for BLE<'a, A> {
+impl<'a, A: kernel::hil::time::Alarm + 'a> kernel::hil::time::Client for BLE<'a, A> {
     // this method is called once the virtual timer has been expired
     // used to periodically send BLE advertisements without blocking the kernel
     fn fired(&self) {
-        self.radio.start_adv();
-        self.configure_periodic_alarm();
+        if self.busy.get() {
+            if self.is_advertising.get() {
+                self.radio.start_adv_tx();
+            } else {
+                self.radio.start_adv_rx();
+            }
+            self.configure_periodic_alarm();
+        }
+    }
+}
+
+// FIXME: Temporary callback function *stolen* from the radio HIL
```

Comment:
@phil-levis _is_ the Radio trait meant to be generic beyond 802.15.4? (if not it's probably not the best name).

@niklasad1 assuming the answer is no, maybe just implement a trait interal to the `nrf51` crate?